src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/dagligReise/beregning/OffentligTransportBeregningValidering.kt

```diff
@@ -0,0 +1,75 @@
+package no.nav.tilleggsstonader.sak.vedtak.dagligReise.beregning
+
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.brukerfeil
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.brukerfeilHvis
+import no.nav.tilleggsstonader.sak.vedtak.dagligReise.domain.BeregningsresultatForPeriode
+import no.nav.tilleggsstonader.sak.vedtak.dagligReise.domain.BeregningsresultatOffentligTransport
+import no.nav.tilleggsstonader.sak.vedtak.domain.InnvilgelseEllerOpphørDagligReise
+import java.time.LocalDate
+
+/**
+ * Validerer at en revurdering av offentlig transport ikke endrer en periode som inneholder dagens dato (er tidligere utbetalt) som enkeltbilletter
+ * til månedskort, da dette kan være til ugunst for søker. Valideringen resulterer i en brukerfeil som veileder saksbehandler må legge inn en ny reise i stedet for å
+ * forlenge en eksisterende periode.
+ *
+ * For hver reise i det nye beregningsresultatet:
+ *  - Sjekker vi om det finnes en tilsvarende reise i forrige beregningsresultat (basert på reiseId).
+ *  - Hvis ja, sjekker om det finnes en periode for dagens dato i både revurderingen og førstegangsbehandlingen.
+ *  - Dersom endringen på perioden endrer billetttype fra enkeltbilletter til månedskort, kastes en veiledende brukerfeil.
+ */
+fun validerRevurdering(
+    beregningsresultat: BeregningsresultatOffentligTransport,
+    tidligsteEndring: LocalDate?,
+    forrigeIverksatteVedtak: InnvilgelseEllerOpphørDagligReise,
+) {
+    brukerfeilHvis(tidligsteEndring == null) {
+        "Kan ikke beregne ytelse fordi det ikke er gjort noen endringer i revurderingen"
+    }
+    val dagensDato = LocalDate.now()
+
+    val nyttBeregningsresultat = beregningsresultat.reiser
+    val forrigeBeregningsresultat = forrigeIverksatteVedtak.beregningsresultat.offentligTransport?.reiser ?: return
+
+    for (reise in nyttBeregningsresultat) {
+        val overlappendeReise = forrigeBeregningsresultat.filter { it.reiseId == reise.reiseId }
+        if (overlappendeReise.isEmpty()) continue
```

Comment:
overlappendeReise er ikke så beskrivende dersom perioden har endret datoer såpass at de ikke lenger er overlappende

```suggestion
        val reiseIForrigeBeregningsresultat = forrigeBeregningsresultat.filter { it.reiseId == reise.reiseId }
        if (reiseIForrigeBeregningsresultat.isEmpty()) continue
```